src/uct/cuda/cuda_copy/cuda_copy_iface.c

```diff
@@ -0,0 +1,243 @@
+    iface_attr->cap.am.max_bcopy        = 0;
+    iface_attr->cap.am.min_zcopy        = 0;
+    iface_attr->cap.am.max_zcopy        = 0;
+    iface_attr->cap.am.opt_zcopy_align  = 1;
+    iface_attr->cap.am.align_mtu        = iface_attr->cap.am.opt_zcopy_align;
+    iface_attr->cap.am.max_hdr          = 0;
+    iface_attr->cap.am.max_iov          = 1;
+
+    iface_attr->latency.overhead        = 10e-6;
+    iface_attr->latency.growth          = 0;
+    iface_attr->bandwidth               = 6911 * 1024.0 * 1024.0;
+    iface_attr->overhead                = 1000;
+    iface_attr->priority                = 0;
+
+    return UCS_OK;
+}
+static unsigned uct_cuda_copy_iface_progress(uct_iface_h tl_iface)
+{
+    uct_cuda_copy_iface_t *iface = ucs_derived_of(tl_iface, uct_cuda_copy_iface_t);
+    unsigned count = 0;
+
+    ucs_queue_iter_t iter;
+    uct_cuda_copy_event_desc_t *cuda_event;
+    cudaError_t result = cudaSuccess;
+
+    ucs_queue_for_each_safe(cuda_event, iter, &iface->pending_event_q, queue)
+    {
+        result = cudaEventQuery(cuda_event->event);
+        if (cudaSuccess == result) {
+            ucs_queue_del_iter(&iface->pending_event_q, iter);
+            cuda_event->comp->func(cuda_event->comp, UCS_OK);
+            ucs_info("Event Done :%p", cuda_event);
+            ucs_mpool_put(cuda_event);
+            count++;
+        }
+    }
+    return count;
+}
+
+static uct_iface_ops_t uct_cuda_copy_iface_ops = {
+    .ep_get_zcopy             = uct_cuda_copy_ep_get_zcopy,
+    .ep_pending_add           = ucs_empty_function_return_busy,
+    .ep_pending_purge         = ucs_empty_function,
+    .ep_flush                 = uct_base_ep_flush,
+    .ep_fence                 = uct_base_ep_fence,
+    .ep_create_connected      = UCS_CLASS_NEW_FUNC_NAME(uct_cuda_copy_ep_t),
+    .ep_destroy               = UCS_CLASS_DELETE_FUNC_NAME(uct_cuda_copy_ep_t),
+    .iface_flush              = uct_base_iface_flush,
+    .iface_fence              = uct_base_iface_fence,
+    .iface_progress_enable    = ucs_empty_function,
+    .iface_progress_disable   = ucs_empty_function,
+    .iface_progress           = uct_cuda_copy_iface_progress,
+    .iface_close              = UCS_CLASS_DELETE_FUNC_NAME(uct_cuda_copy_iface_t),
+    .iface_query              = uct_cuda_copy_iface_query,
+    .iface_get_device_address = (void*)ucs_empty_function_return_success,
+    .iface_get_address        = uct_cuda_copy_iface_get_address,
+    .iface_is_reachable       = uct_cuda_copy_iface_is_reachable,
+};
+
+void uct_cuda_copy_event_desc_init(ucs_mpool_t *mp, void *obj, void *chunk)
+{
+    uct_cuda_copy_event_desc_t *base = (uct_cuda_copy_event_desc_t *) obj;
+    ucs_status_t status;
+    memset(base, 0 , sizeof(*base));
+    status = CUDA_FUNC(cudaEventCreateWithFlags(&(base->event), cudaEventDisableTiming));
+    if (UCS_OK != status) {
+        ucs_error("cudaEventCreateWithFlags Failed");
+    }
+}
+void uct_cuda_copy_event_desc_cleanup(ucs_mpool_t *mp, void *obj)
```

Comment:
you don't call these functions directly by name from anywhere, may be declare it as 'static'?